pkg/remote/aws/s3_bucket_metrics_detail_fetcher.go

```diff
@@ -0,0 +1,38 @@
+package aws
+
+import (
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+	"github.com/cloudskiff/driftctl/pkg/terraform"
+)
+
+type S3BucketMetricsDetailsFetcher struct {
+	reader       terraform.ResourceReader
+	deserializer *resource.Deserializer
+}
+
+func NewS3BucketMetricsDetailsFetcher(provider terraform.ResourceReader, deserializer *resource.Deserializer) *S3BucketMetricsDetailsFetcher {
+	return &S3BucketMetricsDetailsFetcher{
+		reader:       provider,
+		deserializer: deserializer,
+	}
+}
+
+func (r *S3BucketMetricsDetailsFetcher) ReadDetails(res resource.Resource) (resource.Resource, error) {
+	ctyVal, err := r.reader.ReadResource(terraform.ReadResourceArgs{
+		Ty: aws.AwsS3BucketMetricResourceType,
```

Comment:
I'm not sure of the relevance of this remark but why don't we have a variable like in suppliers to call the type of the resource? (supportedType())

Reply:
This variable will be used in enumerators (not in suppliers) to filter resources we do not want without the need to scan them. If the enumerator do not run, then the detail fetcher will not run too as it uses enumerators output. So this is not relevant here